### ISSUE TITLE
Refactor some code to make `lib/archive` usable from other projects

### ISSUE THREAD

**Issue by harikb** [adobe/blackhole#2]
- Remove blackhole "http request" specific items from `lib/archive/file` 
- With those items removed, this library will have the following features

1. For any kind of logging/archiving application, we will have a worker who needs to manage local files and keep feeding data to it.
2. Make sure that a file is externally visible only after it is complete. This is particularly important if the file is compressed.
3. Since there are no "hidden files", we manage this by keeping file under a temporary name and tracking the tasks to flush buffers, close, and rename the file to its final name
4. Rotate the file once it is above a certain size limit. Size cut-off is not part of the library, just that a public method is exposed.
5. Transparently manage compression and decompression - only LZ4 is supported currently.
